packages/babel-cli/src/babel/options.js

```diff
@@ -281,3 +287,24 @@ function collect(value, previousValue): Array<string> {
 
   return previousValue ? previousValue.concat(values) : values;
 }
+
+function getOutDir(): string {
+  const outDirOptions = ["lib", "dist"];
+
+  // check package.json main field if it exists
+  if (fs.existsSync("package.json")) {
+    const packageJson = JSON.parse(fs.readFileSync("package.json"));
+
+    if (packageJson && packageJson.main) {
+      return path.dirname(packageJson.main);
```

Comment:
It's valid for a `package.json#main` to actually be a folder, so this will have issues, I'd suggest that we do

    const basedir = process.cwd();
    const filepath = require.resolve(path.resolve(basedir, packageJson.main));
    return path.dirname(filepath);

so that we resolve the specific file first.

Reply:
Also we may want to skip here if the dirname that is returned is the root of the package and not a subfolder.

Reply:
For this second comment are you saying if `dirname` is the same as the `basepath` then we should just default to something like `lib` or `dist` (instead of writing the output to the root of the package)?